lib/chef/resource/mdadm.rb

```diff
@@ -99,6 +106,37 @@ def raid_device(arg=nil)
         )
       end
 
+      # Can be removed once metadata and chunk defaults are set to nil
+      def user_set_metadata(arg=nil)
+        set_or_return(
+          :exists,
+          arg,
+          :kind_of => [ TrueClass, FalseClass ]
+        )
+      end
+
+      # Can be removed once metadata and chunk defaults are set to nil
+      def user_set_chunk(arg=nil)
+        set_or_return(
+          :exists,
+          arg,
+          :kind_of => [ TrueClass, FalseClass ]
+        )
+      end
+
+      # Can be removed once metadata and chunk defaults are set to nil
+      def after_created
+        metadata_warn_msg = 'The default metadata version of 0.90 will be '\
+          'removed in a future release. This change will allow mdadm to '\
+          'choose the default metadata version.'
+        chunk_warn_msg = 'The default chunk size of 16k will be removed in a '\
+          'future release.  This change will allow mdadm to choose '\
+          'the default chunk size.'
```

Comment:
Do you think that wording may scare people?  This resource doesn't modify any existing raid devices.  So the planned change will only affect new devices created with the resource.  I don't want it to make anyone think that they have to make this change or risk breaking their existing md devices.

What do you think about this wording?
"#{self} default chunk size of 16k will be removed in a future release. To maintain backwards compatibility please explicitly specify the chunk size that you desire on the the mdadm resource if the mdadm default is not desired. This future change will only impact newly created md devices."